### PR TITLE
Bomb Defusal Things

### PR DESCRIPTION
**Pull request by Geevies**
* Added the ability to defuse various bombs with wirecutter-like tools, as well as implementing a specialized bomb-defusal tool, which is not absolutely guaranteed to work.

### CODE REVIEW

**Comment by Geevies** [comment]
!review

File: code/game/objects/items/weapons/landmines.dm [hunk 1]
```diff
@@ -77,9 +75,21 @@
...
 		..()
 
-/obj/item/landmine/attackby(var/obj/item/I, var/mob/user)
+/obj/item/landmine/attackby(obj/item/I, mob/user)
 	..()
-	if(I.force > 10 && deployed)
+	if(deployed && istype(I, /obj/item/wirecutters))
+		var/obj/item/wirecutters/W = I
+		user.visible_message(SPAN_WARNING("\The [user] starts defusing \the [src] with \the [W]." \
+							SPAN_NOTICE("You start defusing \the [src] with \the [W].")))
```
Comment (NonQueueingMatt) [inline code/game/objects/items/weapons/landmines.dm]: Maybe add some flavor to tense moments like these? Could say something like, `[user] starts snipping some wires in \the [W] with \the [src]...` or something.

Comment (Geevies) [reply code/game/objects/items/weapons/landmines.dm]: donezo

File: code/game/objects/items/weapons/tools.dm [hunk 2]
```diff
@@ -145,6 +146,14 @@
...
 	return TRUE
 
+/obj/item/wirecutters/bomb
+	name = "bomb defusal wirecutters"
+	desc = "A tool used to delicately sever the wires used in bomb fuses."
+	icon = 'icons/obj/tools.dmi'
+	icon_state = "cutters"
+	toolspeed = 0.6
+	bomb_defusal_chance = 90 // 90% chance, because the thrill of dying must be kept at all times, duh
+
```
Comment (NonQueueingMatt) [inline code/game/objects/items/weapons/tools.dm]: Considering this is pretty much a general use item I'd love to see this as like a briefcase-full-of-tools sort of deal. Like you open it, you can see a lot of tools in the sprite and then you can just defuse things by clicking on bombs with the open briefcase.

Comment (Geevies) [reply code/game/objects/items/weapons/tools.dm]: talked with you about this in discord now, i think we can do this in the future when we [REDACTED] and implement [REDACTED]

**Comment by Alberyk** [comment]
No, defusing landmines makes zero sense. Also, it would make them far more useless, there is several ways to disarm/explode them already.

**Comment by Geevies** [comment]
How?

**Comment by Alberyk** [comment]
Shoot them, throw a large mob over them or just hit them with something that has the reach of more than one tile. Using a wirecutter to disable a landmine is too much, mostly when you take in consideration that in real life, most landmines are commonly disabled by detoning them.

**Comment by Geevies** [comment]
I don't agree. I think it doesn't make sense that the only way to deal with this is to ignore it or to blow it up. It's area denial, it takes time and every defusal carries a risk of detonation. If you want surefire damage, put more effort into a trap, or put yourself in danger and throw a grenade.

I have found myself frustrated when people try to do EOD roleplay, then have to step on it or do something ridiculous just to get rid of the effortlessly placed mine. Building a wall over it is my go-to move currently.

If you have an alternative way you wish to do this, feel free to open a PR which introduces a more futuristic method of doing this, if you don't like this, and I will gladly close this PR.

**Comment by Doxxmedearly** [comment]
I think this is a good PR. As Geeves said, mines are area denial tools, and making people take the time to disarm them with an EOD getup is pretty cool RP and makes them more effective denial tools due to the time needed to dispose of them.

Detonation and disarming are both tools used for mine clearance, however as we are on a space station I would think that minimizing potential breaches would be a priority and disarming would be preferred to detonation. A futuristic disarming tool could be very cool, too. It does nothing to remove quicker and frankly sillier methods of removing them so I do not agree that it is a nerf.

File: code/modules/assembly/bomb.dm [hunk 3]
```diff
@@ -26,15 +26,14 @@
 	if(istype(W, /obj/item/device/analyzer))
 		bombtank.attackby(W, user)
 		return
-	if(W.iswrench() && !status)	//This is basically bomb assembly code inverted. apparently it works.
+	if((W.iswrench() && !status) || istype(W, /obj/item/wirecutters/bomb))	//This is basically bomb assembly code inverted. apparently it works.
```
Comment (skull132) [inline code/modules/assembly/bomb.dm]: Does this actually work, should it be: `(!status && (W.iswrench() || istype(W, ...)))`

Comment (Geevies) [inline code/modules/assembly/bomb.dm]:
```suggestion
	if(!status && (W.iswrench() || istype(W, /obj/item/wirecutters/bomb)))	//This is basically bomb assembly code inverted. apparently it works.
```

File: code/game/objects/items/weapons/landmines.dm [hunk 4]
```diff
@@ -4,12 +4,11 @@
 	icon = 'icons/obj/grenade.dmi'
 	icon_state = "landmine"
 	throwforce = 0
-	w_class = 3
```
Comment (NonQueueingMatt) [inline code/game/objects/items/weapons/landmines.dm]: This should not have been removed.

**Pull request merged**